core/lib/libtomcrypt/src/pk/ec25519/tweetnacl.c

```diff
@@ -242,6 +242,27 @@ static int tweetnacl_crypto_hash(u8 *out,const u8 *m,u64 n)
   return 0;
 }
 
+static int tweetnacl_crypto_hash_ctx(u8 *out,const u8 *m,u64 n,const u8 *ctx,u32 cs)
+{
+  unsigned long len;
+  int err;
+  u8 buf[512];
```

Comment:
in tweetnacl they do all allocations on stack, so we followed their a style (though other allocations in this library are considerably smaller).

As for 512 size considerations are:
255 (context size) + 32 context prefix + 1 flag + 1 ctx len
PREFIX=b"SigEd25519 no Ed25519 collisions" so minimal allocation of 289 seems to be required.
But there are another algorithms that might require context prefix. So it was rounded up to 512.

Reply:
@etienne-lms I also thought it could be worth it, but apparently tweetnacl itself is already very stack-heavy.

Each invocation of `tweetnacl_crypto_hash[_ctx]()` inside tweetnacl originates from a function that also calls other functions. Those other functions require way more stack than 512 bytes, ergo that stack size is already required.

If I counted correctly the call-stack of `tweetnacl_crypto_sign(): scalarbase() -> scalarmult() -> add() -> M() -> car25519()` requires those stack variables:

```
typedef i64 gf[16];
  int i; //4 or 8
  i64 c; //8
  i64 i,j,t[31]; //264
  gf a,b,c,d,t,e,f,g,h; //1152
  u8 b; //1
  int i; //4 or 8
  gf q[4]; //512
// max. sum of the above: 1953 bytes
```

Reply:
I still liked the idea so there's the last commit of https://github.com/libtom/libtomcrypt/pull/598

Reply:
Patch backported as a fixup: [ff58986](https://github.com/OP-TEE/optee_os/pull/5486/commits/ff58986367bc764f3c51e78ab8c60a0e4920fa83)